modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/info/collection/provider/SXPBlueprintInfoCollectionProvider.java

```diff
@@ -0,0 +1,254 @@
+								CategoriesInfoFilter.class);
+
+						if ((categoriesInfoFilter != null) &&
+							!ArrayUtil.isEmpty(
+								categoriesInfoFilter.getCategoryIds())) {
+
+							long[] categoryIds = ArrayUtil.append(
+								categoriesInfoFilter.getCategoryIds());
+
+							categoryIds = ArrayUtil.unique(categoryIds);
+
+							searchContext.setAssetCategoryIds(categoryIds);
+						}
+
+						TagsInfoFilter tagsInfoFilter =
+							collectionQuery.getInfoFilter(TagsInfoFilter.class);
+
+						if ((tagsInfoFilter != null) &&
+							!ArrayUtil.isEmpty(tagsInfoFilter.getTagNames())) {
+
+							String[] tagNames = ArrayUtil.append(
+								tagsInfoFilter.getTagNames());
+
+							tagNames = ArrayUtil.unique(tagNames);
+
+							searchContext.setAssetTagNames(tagNames);
+						}
+
+						KeywordsInfoFilter keywordsInfoFilter =
+							collectionQuery.getInfoFilter(
+								KeywordsInfoFilter.class);
+
+						if (keywordsInfoFilter != null) {
+							searchContext.setKeywords(
+								keywordsInfoFilter.getKeywords());
+						}
+
+						searchContext.setAttribute(
+							"search.experiences.blueprint.id",
+							sxpBlueprint.getSXPBlueprintId());
+						searchContext.setLocale(serviceContext.getLocale());
+
+						searchContext.setAttribute(
+							"search.experiences.ip.address",
+							serviceContext.getRemoteAddr());
+
+						ThemeDisplay themeDisplay =
+							serviceContext.getThemeDisplay();
+
+						searchContext.setAttribute(
+							"search.experiences.scope.group.id",
+							themeDisplay.getScopeGroupId());
+
+						searchContext.setTimeZone(serviceContext.getTimeZone());
+						searchContext.setUserId(serviceContext.getUserId());
+					}
+				);
+
+			SearchResponse searchResponse = _searcher.search(
+				searchRequestBuilder.build());
+
+			return InfoPage.of(
+				_assetHelper.getAssetEntries(searchResponse.getSearchHits()));
+		}
+		catch (NoSuchSXPBlueprintException noSuchSXPBlueprintException) {
+			_log.error(noSuchSXPBlueprintException);
+		}
+		catch (Exception exception) {
+			_log.error("Unable to get asset entries", exception);
+		}
+
+		return InfoPage.of(
+			Collections.emptyList(), collectionQuery.getPagination(), 0);
+	}
+
+	// TODO Implement SXPBlueprntsOptionsPortlet Selector screen
+
+	@Override
+	public InfoForm getConfigurationInfoForm() {
+		return InfoForm.builder(
+		).infoFieldSetEntry(
+			InfoField.builder(
+			).infoFieldType(
+				TextInfoFieldType.INSTANCE
+			).namespace(
+				StringPool.BLANK
+			).name(
+				"sxpBlueprintExternalReferenceCode"
+			).labelInfoLocalizedValue(
+				InfoLocalizedValue.localize(getClass(), "blueprint")
+			).localizable(
+				false
+			).build()
+		).build();
+	}
+
+	@Override
+	public String getLabel(Locale locale) {
+		return _language.get(locale, "blueprint");
+	}
+
+	@Override
+	public List<InfoFilter> getSupportedInfoFilters() {
+		return Arrays.asList(
+			new CategoriesInfoFilter(), new KeywordsInfoFilter(),
+			new TagsInfoFilter());
```

Comment:
Check with Echo why the Tags filter does not show up on screen.

Reply:
From Lourdes @lfbesada:

https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2228715772/LPS-167949+Support+filtering+by+tags+in+the+collection+display+filter

https://issues.liferay.com/browse/LPS-167949

Maybe it was intentionally limited to only work with Collections, but not with Collection Providers. Need to check with Julia/Pablo.